chapter.md

```diff
@@ -0,0 +1,966 @@
+"This is an observable."
+We don't have to use it,
+but we find it helps make code easier to understand.)
+
+We can now go ahead and create specific rules for specific fields of our form
+using a helper function called `isBetweenNumber` that does exactly what you'd expect:
+
+```ts
+const humanAgeValid = isBetweenNumber(14, 40);
+const elfAgeValid = isBetweenNumber(80, 800);
+const tieflingAgeValid = isBetweenNumber(35, 53);
+
+const bioSummarySelector = createSelector(
+  characterFormSelector,
+  ({bioSummary}: ICharacter) => bioSummary
+);
+
+const ageValidationSelector = createSelector(
+  bioSummarySelector,
+  (bioSummary: IBioSummary) => {
+    switch (bioSummary.race) {
+      case 'Human': return humanAgeValid;
+      case 'Elf': return elfAgeValid;
+      case 'Tiefling': return tieflingAgeValid;
+    }
+  }
+);
+
+export const isAgeValidSelector = createSelector(
+  bioSummarySelector,
+  ageValidationSelector,
+  ({age}, isAgeValid) => isAgeValid(age)
+);
+```
+
+Once we have a pile of validation rules for our fields,
+we can chain those selectors together like this:
+
+```ts
+const isFormValidSelector = createSelector(
+  isAgeValidSelector,
+  isNameValidSelector,
+  (ageValid, nameValid) => ageValid && nameValid
+);
+```
+
+In larger applications,
+we can and should go further
+and make selectors out of reusable validation functions:
+
+```ts
+const maxStringLengthValidation = (value: string, max: number) =>
+  value.length < max;
+
+const minStringLengthValidation = (value: string, min: number) =>
+  value.length > min;
+
+const isNameValidSelector = createSelector(
+  createFormFieldSelector(['character', 'name']),
+  (name: string) => maxStringLengthValidation(name, 50)
+    && minStringLengthValidation(name, 3)
+);
+```
+
+Once we're doing this,
+we will probably also want to create a generic function
+to check multiple validations:
+
+```ts
+const isValid = (...validators): any =>
+  (arg: any) => isNil(find(val => !val(arg), validators));
+```
+
+and then go back and create smaller validation functions:
+
+```ts
+const maxNumberValidation = (max: number) =>
+  (value: number) => value < max;
+
+const maxStringLengthValidation = (max: number) =>
+  (value: string) => value.length < max;
+
+const minStringLengthValidation = (min: number) =>
+  (value: string) => value.length > min;
+```
+
+so that we can pipe our validation rules into the generic `isValid`:
+
+```ts
+const createFormFieldSelector = (fieldPath: string[]) => createSelector(
+  formStateSelector,
+  (form: IForm) => path(fieldPath, form)
+);
+
+const isNameValidSelector = createSelector(
+  createFormFieldSelector(['character', 'name']),
+  isValid(
+    maxStringLengthValidation(50),
+    minStringLengthValidation(3),
+  ),
+);
+```
+
+We can make our application more user-friendly
+by subscribing to the selectors we've made and using them in our template:
+
+```ts
+@select(isAgeValidSelector)
```

Comment:
`@select` appears slightly earlier here but also undocumented... Worth mentioning what this does. This is from `angular-redux` I believe and sets up a subscription. Our previous example of subscriptions in the form component show the long form, and this is just a shorthand IIRC. @danielfigueiredo please confirm?